runtimes/eoapi/stac/eoapi/stac/client.py

```diff
@@ -383,6 +409,9 @@ async def all_collections(
         **kwargs,
     ) -> Collections:
         collections = await super().all_collections(request, *args, **kwargs)
+        if self.titiler_endpoint:
```

Comment:
you should be able to get the titiler_endpoint from `request.app.settings.titiler_endpoint` I think